### PR TITLE
PHP 8: Code Review `Navigation`

### PR DESCRIPTION
**Pull request by mjansenDatabay**
Hi @alex40724 ,

I completed the review of the `Navigation` component.

Results:
- [x] The code style is `PSR-2 + X` compliant
- [x] No usage of `$_GET` / `$_POST` / `$_REQUEST` / `$_SESSION`
- [x] There is a `Unit Test Suite` with at least one unit test
- [x] The unit tests pass
- [x] External libraries are compatible with PHP 8 (if relevant)
- [ ] There are no serious `PhpStorm Code Inspection` issues left
- [ ] The types are fully documented (PhpDoc) or explict PHP types are used (type hints, return types, typed properties)

I also fixed some trivial issues reported by my inspection profile.

--

Actions needed:

- [ ] Please check my inline comments. Some of them might be just information, some require an action. You can use `TODO PHP8-REVIEW` when searching.

Best regards,
@mjansenDatabay

### CODE REVIEW

File: Services/Navigation/classes/class.ilNavigationHistory.php [hunk 1]
```diff
@@ -146,7 +146,7 @@ public function getItems() : array
                             $tree->isInTree($rec["ref_id"]) &&
                             (
                                 !$objDefinition->isPluginTypeName($rec["type"]) ||
-                                $component_repository->getPluginById($it["type"])->isActive()
+                                $component_repository->getPluginById($it["type"])->isActive()// TODO PHP8-REVIEW $it is based on the laste iteration above. I suggest to use an ecplit new variable for this case.
```
Comment (mjansenDatabay) [inline Services/Navigation/classes/class.ilNavigationHistory.php:149]: `$it ` is based on the laste iteration above. I suggest to use an ecplit new variable for this case.

File: Services/Navigation/classes/class.ilNavigationHistory.php [hunk 2]
```diff
@@ -131,37 +131,33 @@ public function getItems() : array
...
-                            (
-                                !$objDefinition->isPluginTypeName($rec["type"]) ||
-                                $component_repository->getPluginById($it["type"])->isActive()
-                            )
-                        ) {
-                            $link = ($rec["goto_link"] != "")
+                    if ($cnt <= 10 && !isset($items[$rec["ref_id"] . ":" . $rec["sub_obj_id"]]) && $tree->isInTree($rec["ref_id"]) &&
+                        (
+                            !$objDefinition->isPluginTypeName($rec["type"]) ||
+                            $component_repository->getPluginById($it["type"])->isActive()// TODO PHP8-REVIEW $it is based on the laste iteration above. I suggest to use an explicit new variable for this case.
```
Comment (mjansenDatabay) [inline Services/Navigation/classes/class.ilNavigationHistory.php:147]: `$it` is based on the laste iteration above. I suggest to use an explicit new variable for this case

**Comment by alex40724** [comment]
Thanks. I will take care of the todo.

File: Services/Navigation/classes/class.ilNavigationHistory.php (hunk 2 above)
Comment (alex40724) [reply Services/Navigation/classes/class.ilNavigationHistory.php:147]: This made me puzzling, its a wrong variable being accessed. But a trunk only issue, see https://github.com/ILIAS-eLearning/ILIAS/commit/4786c2e55dda9532240ed3510480eb08dae3a684#diff-c3c5149c3990e61a2f036839a286269837736dff68894bb855fc6dc0a5e89090

**Comment by alex40724** [comment]
Done.

**Pull request merged**